<a href="https://colab.research.google.com/github/canunz/LIBRARY-django/blob/main/KNN.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Clasificación con KNN

Modelo de clasificación utilizando knn sobre datos de desempeño en partidas.

In [ ]:
# 📥 Subir archivo desde tu equipo (funciona en Google Colab)
from google.colab import files
uploaded = files.upload()

import pandas as pd
import io

df = pd.read_csv(io.BytesIO(uploaded['datos_exportados.csv']))

Saving datos_exportados.csv to datos_exportados.csv


In [ ]:
# 🔍 Importar librerías necesarias
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
import seaborn as sns
import matplotlib.pyplot as plt

In [ ]:
# ✅ Selección de variables
features = ['TimeAlive', 'TravelledDistance', 'RLethalGrenadesThrown', 'RNonLethalGrenadesThrown', 'PrimaryAssaultRifle', 'PrimarySniperRifle', 'PrimaryHeavy', 'PrimarySMG', 'PrimaryPistol', 'RoundKills', 'RoundAssists', 'RoundHeadshots', 'RoundFlankKills', 'RoundStartingEquipmentValue', 'TeamStartingEquipmentValue', 'MatchKills', 'MatchFlankKills', 'MatchAssists', 'MatchHeadshots']
target = 'Survived'

X = df[features]
y = df[target]

# 🔀 Separar en entrenamiento y prueba
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=666)

# ⚖️ Escalar características
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)


In [ ]:

from sklearn.neighbors import KNeighborsClassifier

model = KNeighborsClassifier()
model.fit(X_train_scaled, y_train)
y_pred = model.predict(X_test_scaled)

print("Accuracy:", accuracy_score(y_test, y_pred))
print("\nMatriz de Confusión:")
print(confusion_matrix(y_test, y_pred))
print("\nReporte de Clasificación:")
print(classification_report(y_test, y_pred))

Accuracy: 0.6311877006251055

Matriz de Confusión:
[[2385  947]
 [1236 1351]]

Reporte de Clasificación:
              precision    recall  f1-score   support

           0       0.66      0.72      0.69      3332
           1       0.59      0.52      0.55      2587

    accuracy                           0.63      5919
   macro avg       0.62      0.62      0.62      5919
weighted avg       0.63      0.63      0.63      5919



###🧠 Conclusión del Modelo

🔎 Precisión global (accuracy) del modelo es 0.631, lo que significa que clasifica correctamente aproximadamente el 63.1% de los casos.

🧮 Matriz de Confusión:

Clase 0: Bien identificada (Recall = 0.72)

Clase 1: Recall bajo (0.52), aunque mejor que en regresión logística.

⚠️ Desempeño desigual: El modelo aún tiene dificultades para predecir correctamente la clase 1, lo que puede ser crítico si esa clase representa un evento importante.

📉 Macro F1-score: 0.62, lo que sugiere un desempeño moderado en ambas clases, pero con amplio espacio para mejora.

📌 Sensibilidad al escalado: Como modelo basado en distancias, KNN requiere escalado. Esto fue correctamente abordado con StandardScaler, pero puede mejorar usando optimización de hiperparámetros (k, distancia, etc.).

